pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -1422,10 +1431,7 @@ func convertGateways(r ConfigContext) ([]config.Config, map[parentKey]map[k8s.Se
 		reportGatewayCondition(obj, gatewayConditions)
 	}
 	// Insert a parent for Mesh references.
-	gwMap[parentKey{
-		Kind: meshGVK,
-		Name: "istio",
-	}] = map[k8s.SectionName]*parentInfo{
+	gwMap[meshParentKey] = map[k8s.SectionName]*parentInfo{
```

Comment:
Should we still have this? In 1426 there is no `Mesh` parent anymore, just Service

Reply:
I figured that using it internally was a fairly clean way to fit Service-type parents into the existing code, without too much restructuring, and maybe a future GEP will add a mesh resource anyway? On line 634, I use this "virtual gw" for all Service parents:
```
		gk := ir
		if ir.Kind == gvk.Service {
			gk = meshParentKey
		}
```

Reply:
Hmm.. it was a bit confusing to me since in some places its a Service and in others its a Mesh (`parent.Kind == gvk.Service {` for example)

I get we later swap it out so it works, just wasn't obvious to me.